code/modules/mob/living/carbon/xenomorph/abilities/general_powers.dm

```diff
@@ -555,6 +551,15 @@
 		to_chat(X, SPAN_XENOWARNING("This is too close to a fruit!"))
 		return
 
+	if(istype(alien_weeds, /obj/effect/alien/weeds/node))
+		to_chat(X, SPAN_NOTICE("You start uprooting the node so you can put the resin hole in its place..."))
+		if(!do_after(X, 1 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC, target, INTERRUPT_ALL))
+			return
+		var/obj/effect/alien/weeds/the_replacer = new /obj/effect/alien/weeds(alien_weeds.loc)
+		the_replacer.hivenumber = X.hivenumber
```

Comment:
use set_hive_data and then update the icon and whatever to make sure the sprites are correct

Reply:
`Set_hive_data` sets the sprites and name and not much else, for whatever reason I have to manually define the `hivenumber` and `linked_hive` as well.